### PR TITLE
Fixed #23 - [mariadb] Set /data as base folder for mysql logs. Fixed default EN…

### LINKED ISSUE #23: MariaDB can’t be launched (Supervisor error)
Run command :
```
docker run -d --name="xxxxx_DB" --volumes-from="xxxxx_DBDATA" -e "MARIADB_USER=xxxxx" -e "MARIADB_PASS=xxxxxxx" --restart=always maxexcloo/mariadb
```

MariaDB logs :
```
2016-09-26 12:09:47,104 INFO exited: mariadb (exit status 0; expected)
/usr/lib/python2.7/dist-packages/supervisor/options.py:296: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  'Supervisord is running as root and it is searching '
2016-09-26 12:23:26,874 CRIT Supervisor running as root (no user in config file)
2016-09-26 12:23:26,874 WARN Included extra file "/etc/supervisor/conf.d/mariadb.conf" during parsing
Unlinking stale socket /var/run/supervisor.sock
```

```
2016-09-26 14:17:38,386 CRIT Server 'unix_http_server' running without any HTTP authentication checking
```

My docker version :
```
➜  docker -v
Docker version 1.12.1, build 23cf638
```

### PR DESCRIPTION
**Pull request by ambroisemaupate**
- Set `/data` as base folder for every mysql logs, fixing initial startup issue.  
- `chown` mysql socket (`/run/mysql`) which prevented mysql daemon to listen requests.
- Fixed default ENV vars syntax in Dockerfile. 
- Create a default database named after user.

### CODE REVIEW

File: services/mariadb/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 FROM maxexcloo/debian:latest
 MAINTAINER Max Schaefer <max@excloo.com>
-ENV MARIADB_USER docker \
-	MARIADB_PASS docker
```
Comment (ambroisemaupate) [inline services/mariadb/Dockerfile]: This syntax generated a `MARIADB_USER` var with `docker MARIADB_PASS docker` content. Found it doing a `echo $MARIADB_USER` while debug my mariadb container.

File: services/mariadb/config/init/mariadb [hunk 2]
```diff
@@ -1,9 +1,16 @@
-if [ ! "$(ls -A /data)" ]; then
+if [ ! "$(ls -A /data/aria_log_control)" ]; then
```
Comment (ambroisemaupate) [inline services/mariadb/config/init/mariadb]: With your new `debian` image, `/data` folder is no longer empty (you create `/data/log` folder in init loop), so the script was never executing mysql provision stuff.

File: services/mariadb/config/init/mariadb [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+	mkdir -p /data/log
+	chown -R core:core /data/log
 	mysql_install_db --datadir=/data --user=core
 	mysqld_safe --skip-grant-tables &
 	while [ ! -e /run/mysqld/mysqld.sock ]; do
 		inotifywait -e create -q /run/mysqld/
 	done
-	mysql -u root -e "FLUSH PRIVILEGES; CREATE USER '$MARIADB_USER'@'%' IDENTIFIED BY '$MARIADB_PASS'; GRANT ALL PRIVILEGES ON *.* TO '$MARIADB_USER'@'%' WITH GRANT OPTION;"
+	mysql -u root -e "FLUSH PRIVILEGES; CREATE USER '${MARIADB_USER}'@'%' IDENTIFIED BY '${MARIADB_PASS}'; GRANT ALL PRIVILEGES ON *.* TO '${MARIADB_USER}'@'%' WITH GRANT OPTION;"
+	mysql -u root -e "CREATE DATABASE IF NOT EXISTS ${MARIADB_USER};"
```
Comment (ambroisemaupate) [inline services/mariadb/config/init/mariadb]: I think it’s convinient to provision a default database named after user.

File: services/mariadb/config/init/mariadb (hunk 2 above)
Comment (maxexcloo) [inline services/mariadb/config/init/mariadb]: Good catch!

**Pull request merged**